### PR TITLE
User login and creation

### PR DESCRIPTION
**Pull request by dylancastillo**
Enable a user to login, give servlets a way to check whether the user is logged in, and create an user entity which stores the groups the user belongs in.

### CODE REVIEW

File: src/main/java/com/google/sps/data/User.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.google.sps.data;
+
+import java.util.HashSet;
+
+public final class User {
+  private final String email;
+  private final HashSet<Long> groups;
```
Comment (vlop11) [inline src/main/java/com/google/sps/data/User.java]: it looks like in other parts of the application we're using a string for the groupId. are you planning on parsing it out into a long here?

Comment (vlop11) [reply src/main/java/com/google/sps/data/User.java]: nvm I just saw that you're using Long.valueof() :)

File: src/main/java/com/google/sps/servlets/GroupsServlet.java [hunk 2]
```diff
@@ -96,4 +124,28 @@ private String createGroup(String name, String description, String image){
...
+      HashSet<Long> groups = (HashSet<Long>) user.getProperty(GROUPS_KEY);
+      groups.add(Long.valueOf(groupId));
+      user.setProperty(GROUPS_KEY, groups);
+
+      datastore.put(user);
+    } catch (EntityNotFoundException newUser) {
+      Entity user = new Entity(USER_KIND, email);
+
+      user.setProperty(USER_EMAIL_PROPERTY, email);
+      user.setProperty(GROUPS_KEY, new HashSet<Long>());
```
Comment (vlop11) [inline src/main/java/com/google/sps/servlets/GroupsServlet.java]: I'm not entirely sure on the flow you were going for (so correct me if I'm wrong), but would it be possible to create a new user if they don't already exist, and add them to the group in this same method? for example, on the UI there's a button called "JOIN" for each group. if that  button is pressed, would there even be a possibility that the user does not exist in datastore?

File: src/main/java/com/google/sps/data/User.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+  public User(String email) {
+    this.email = email;
+    this.groups = new HashSet<>();
```
Comment (sporkert) [inline src/main/java/com/google/sps/data/User.java]: You can do this when you instantiate the variable above since it doesn't depend on any passed-in data

**Review by sporkert — APPROVED** [review]

**Review by vlop11 — APPROVED** [review]

**Pull request merged**